test/gtest/ucs/test_log.cc

```diff
@@ -118,21 +118,28 @@ class log_test : public ucs::test {
 
     bool do_grep(const std::string &needle) {
         unsigned num_retries = 0;
+        std::string cmd_str = "";
+        int errno_val = 0;
 
         while (num_retries++ < GREP_RETRIES) {
             /* if this is the last retry, allow printing the grep output */
             std::string grep_cmd = ucs_likely(num_retries != GREP_RETRIES) ?
                                    "grep -q" : "grep";
-            std::string cmd_str = grep_cmd + " '" + needle + "' " +
-                                  template_grep_name;
+            cmd_str = grep_cmd + " '" + needle + "' " + template_grep_name;
             int ret = system(cmd_str.c_str());
             if (ret == 0) {
                 return true;
             }
 
+            errno_val = errno;
```

Comment:
just thought it may be misleading in case the errno is not set (you will see some old value)

Reply:
> just thought it may be misleading in case the errno is not set (you will see some old value)

`errno` value has to be set since this is error handling for `system()`

Reply:
please print `ret` as well to distinguish different failure cases